File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.ts
function numberOfSpecialChars(word: string): number {
    const lowerSet = new Set<string>();
    const upperSet = new Set<string>();

    // 소문자와 대문자를 각각의 Set에 저장
    for (const char of word) {
        if (char >= 'a' && char <= 'z') {
            lowerSet.add(char);
        } else {
            upperSet.add(char.toLowerCase());
        }
    }

    let count = 0;
    // lowerSet에 존재하면서 upperSet에도 존재하는 문자 카운트
    for (const char of lowerSet) {
        if (upperSet.has(char)) {
            count++;
        }
    }

    return count;
}

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.go
func numberOfSpecialChars(word string) int {
    // Arrays to track presence of lowercase and uppercase letters
    lower := make([]bool, 26)
    upper := make([]bool, 26)
    
    for _, char := range word {
        if char >= 'a' && char <= 'z' {
            lower[char-'a'] = true
        } else if char >= 'A' && char <= 'Z' {
            upper[char-'A'] = true
        }
    }
    
    count := 0
    for i := 0; i < 26; i++ {
        if lower[i] && upper[i] {
            count++
        }
    }
    
    return count
}